### PR TITLE
Mean time recovery

### PR DESCRIPTION
**Pull request by YoDaMa**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-node/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

**Comment by anthonyvercolano** [comment]
> #!/usr/bin/env node

What is this?

---
Refers to: sdklab/longhaultests/src/iothub_longhaul.ts:1 in 84bae55. [](commit_id = 84bae551b204e3443bff401f25f6b817b30a003b, deletion_comment = False)

File: sdklab/mean_time_recovery/aedes_server.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+//     server.close();
+//     debug('SERVER CLOSED...')
+//     setTimeout(() => {
+//       debug('RECONNECTING SERVER...')
+//       server.listen(port, function () {
+//         debug(`server reconnected and listening on port ${port}`);
+//       });
+//     }, 10000);
+//   }
+// })
```
Comment (anthonyvercolano) [inline sdklab/mean_time_recovery/aedes_server.js:60]: You wanted this commented out?

File: sdklab/mean_time_recovery/device.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  // setInterval(() => {
+  //   const message = generateMessage();
+  //   client.sendEvent(message, (e) => {
+  //     if (e) {
+  //       console.error('Error on sendEvent: ', e);
+  //     } else {
+  //       debug('Message sent successfully')
+  //     }
+  //   })
+  // }, 2000);
```
Comment (anthonyvercolano) [inline sdklab/mean_time_recovery/device.js:59]: Commented?

File: sdklab/mean_time_recovery/orchestrator.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
+const { fork } = require('child_process');
+const { stderr, mainModule } = require('process');
+const { chunk } = require('lodash');
+const fs = require('fs');
+const debug = require('debug')('orchestrator')
+const writeStream = fs.createWriteStream('dataOutput.log'); // this just clears the file essentially...
+
+function wait (ms) {
+    return new Promise ((resolve, reject) => {
+        setTimeout(resolve, ms);
```
Comment (anthonyvercolano) [inline sdklab/mean_time_recovery/orchestrator.js:10]: Why did we switch to 4 char indent?

File: sdklab/mean_time_recovery/package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "mean_time_recovery",
+  "version": "1.0.0",
+  "description": "metric analysis of mean time to recovery for device code",
+  "main": "ochestrator.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (anthonyvercolano) [inline sdklab/mean_time_recovery/package.json:10]: Wha?

File: sdklab/mean_time_recovery/orchestrator.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            }
+            childDevice.on('message', callback);
+            childDevice.send({ sendPingMessage: true })
+            setTimeout(() => {
+                childDevice.removeListener('message', callback);
+                rej();
+            }, 5000);
+        }); // wait for 1 seconds
+        debug('killing childServer')
+        childServer.kill();
```
Comment (anthonyvercolano) [inline sdklab/mean_time_recovery/orchestrator.js:78]: Seriously.  This looks creepy.  Can we do better?

File: sdklab/mean_time_recovery/device.js (hunk 2 above)
Comment (YoDaMa) [reply sdklab/mean_time_recovery/device.js:59]: yes I'll remove.

**Comment by anthonyvercolano** [comment]
Overall comment for the MTR.  For the overall scheme about what is going on and how each part works is not documented very well.

Internally to each piece should be better documented about what it's individually trying to do to accomplish the overall task.

A much more extensive readme covering would be nice.

**Review by anthonyvercolano — APPROVED** [review]
:shipit:

File: sdklab/mean_time_recovery/orchestrator.js (hunk 5 above)
Comment (YoDaMa) [reply sdklab/mean_time_recovery/orchestrator.js:78]: lol sorry this is just the nomenclature instead of Master/Slave

**Pull request merged**